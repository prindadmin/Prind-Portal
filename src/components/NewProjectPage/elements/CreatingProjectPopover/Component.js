import React, { Component } from 'react'
//import PropTypes from 'prop-types'

import PopOverHandler from '../../../common/popOverHandler'

import {
  Spinner,
  Intent,
} from '@blueprintjs/core'

import * as strings from '../../../../data/Strings'

export class Element extends Component {
  static propTypes = {
  }

  getContent = () => {
    return (
      <React.Fragment>
        {strings.PROJECT_CREATION_BILLING_WARNING}
        <Spinner size='150' intent={Intent.DANGER}/>
      </React.Fragment>
    )
  }

  render () {
    return (
      <PopOverHandler>
        <div id='popup-greyer' onClick={(e) => e.stopPropagation()}>
          <div id='project-creation-popover'>
            <div id='popup-box'>
              <div className='project-creation-popover-container' onClick={(e) => e.stopPropagation()}>
                <div className='element-title'>
                  {strings.PROJECT_CREATION_IN_PROGRESS}
                </div>
                <div className='element-description'>
                  {this.getContent()}
                </div>
              </div>
            </div>
          </div>
        </div>
      </PopOverHandler>
    )
  }

}

export default Element
