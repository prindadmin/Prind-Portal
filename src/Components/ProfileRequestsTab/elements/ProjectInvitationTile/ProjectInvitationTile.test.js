import React from 'react';
import { shallow } from 'enzyme';

import Component from './ProjectInvitationTile'
//import * as Strings from '../../../../Data/Strings'

var props = {
  requestDetails: {
    projectId: "123",
    projectName: "Test Project 123",
    requestedBy: "bbbb-bbbb-bbbbbbbb-bbbb-bbbb",
    requestedAt: "2021-03-30T08:13:45",
    roleId: "client",
    requestedByUser: "Test Testerson",
  },
  respondToProjectInvitation: jest.fn(),
}


it('should render', () => {
  const component = shallow(<Component {...props} />);
  expect(component).toMatchSnapshot();
});


it('should render and accept invitation', () => {
  const mockRespondToProjectInvitation = jest.fn()
  props.respondToProjectInvitation = mockRespondToProjectInvitation

  const component = shallow(<Component {...props} />);
  component.find('#accept-button').simulate('click')
  expect(mockRespondToProjectInvitation).toHaveBeenCalledWith("123", true)
});


it('should render and reject invitation', () => {
  const mockRespondToProjectInvitation = jest.fn()
  props.respondToProjectInvitation = mockRespondToProjectInvitation

  const component = shallow(<Component {...props} />);
  component.find('#reject-button').simulate('click')
  expect(mockRespondToProjectInvitation).toHaveBeenCalledWith("123", false)
});
