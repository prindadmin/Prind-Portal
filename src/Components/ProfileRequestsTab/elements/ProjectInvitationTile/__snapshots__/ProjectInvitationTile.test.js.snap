// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  id="project-invitation-request"
>
  <h4
    className="bp3-heading"
  >
    Project Invitation
  </h4>
  <div
    className="bp3-heading"
  >
    Project Name: Test Project 123
  </div>
  <div
    className="row button-row"
  >
    <Blueprint3.Button
      icon="tick"
      id="accept-button"
      intent="primary"
      onClick={[Function]}
      text="Accept"
    />
    <Blueprint3.Button
      icon="cross"
      id="reject-button"
      intent="danger"
      onClick={[Function]}
      text="Reject"
    />
  </div>
</div>
`;
